src/core/agent/agent.types.ts

```diff
@@ -49,11 +49,37 @@ type ConnectionNoteDetails = {
   message: string;
 };
 
+type IpexMessages = {
+  exn: {
+    v: string;
+    t: string;
+    d: string;
+    i: string;
+    p: string;
+    dt: string;
+    r: string;
+    q: any;
+    a: any;
+    e: any;
+  };
+  pathed: {
+    acdc: string;
+    iss: string;
+    anc: string;
+  };
+};
+
+type IpexMessageDetails = {
+  id: string;
+  content: IpexMessages;
+};
+
 type ConnectionNoteProps = Pick<ConnectionNoteDetails, "title" | "message">;
 
 interface ConnectionDetails extends ConnectionShortDetails {
   serviceEndpoints?: string[];
   notes?: ConnectionNoteDetails[];
+  linkedIpexMessages?: IpexMessageDetails[];
```

Comment:
Hmm, we should be exposing this in the `history`, not directly in `ConnectionDetails`